WcaOnRails/app/controllers/api/internal/v1/payment_controller.rb

```diff
@@ -0,0 +1,64 @@
+# frozen_string_literal: true
+
+class Api::Internal::V1::PaymentController < Api::Internal::V1::ApiController
+  def init
+    return json: { error: "Missing fields" } unless params["attendee_id"].present? && params["amount"].present?
+
+    holder = AttendeePaymentRequest.new(attendee_id: params["attendee_id"])
+    competition_id, user_id = holder.competition_and_user_id
+    amount = params["amount"].to_i
+
+    competition = Competition.find(competition_id)
+    user = User.find(user_id)
```

Comment:
How do we report errors if either isn't found? `Model.find` just returns `nil`, whereas `Model.find!` throws an error

Reply:
I did,  `render json: { error: "Registration not found" }, status: :not_found unless competition.present? && user.present?`
because this way we don't need to rescue the error. Not sure what is more 'rails-y'